### DISCUSSION TITLE
Package service provider not found when trying HTTP GET

### DISCUSSION THREAD

**Discussion by jmares79** [Jeroen-G/laravel-packager#153] [Q&A]
Hi, I created a package following a Laracasts tutorial. The package looks fantastic! But somehow, even following the process by the step I receive the following error when trying to test it:

`include(C:\laragon\www\package-creator\vendor\composer/../laracasts/widget/src/WidgetServiceProvider.php)` 

Steps performed:

1. I created a new Laravel project, called "package-creator" and added your library via the docs
2. I created a new package via `php artisan packager:new Laracasts Widget` command. Package and structure was created ok
3. I added a route in `web.php` like so:  

```
Route::get('/', function () {
    return 'hello';
//    return Widget::test();
});
```
4. I executed `composer require laracasts/widget` in order to create a symlink in the `vendor/laracasts` to `widget`

And when loading the page/route in the browser, I receive the mentioned error.

I tried with `composer dump-autoload` to no avail

Is there any other step I'm missing?

Kind regards, Jose

**Comment by Jeroen-G** [comment]
Hi! Cool that you want to use my package!

I have a few clarifying questions:

1. Does this file exist?

`C:\laragon\www\package-creator\vendor\laracasts\widget\src\WidgetServiceProvider.php`

2. Can you post your composer.json?
3. When/where does the error occur?
    - edit: I just read when you open the website. Does it also happen when you run `php artisan`?

**Reply by jmares79** [reply]
Hi, thanks a lot for answering. And thanks for creating the package as well!!

Answering your questions:

1. Yes, the file exists. I'm in a Windows machine with a WSL console, and there's a symlink from `/mnt/c/laragon/www/package-creator/vendor/laracasts`  like so ` widget -> /mnt/c/laragon/www/package-creator/packages/Laracasts/Widget/`

2. `composer.json` is:

```
{
    "name": "laracasts/widget",
    "description": ":package_description",
    "license": "MIT",
    "homepage": "https://github.com/laracasts/widget",
    "keywords": ["Laravel", "Widget"],
    "require": {
        "illuminate/support": "~8"
    },
    "require-dev": {
        "phpunit/phpunit": "~9.0",
        "orchestra/testbench": "~9"
    },
    "autoload": {
        "psr-4": {
            "Laracasts\\Widget\\": "src/"
        }
    },
    "autoload-dev": {
        "psr-4": {
            "Laracasts\\Widget\\Tests\\": "tests"
        }
    },
    "extra": {
        "laravel": {
            "providers": [
                "Laracasts\\Widget\\WidgetServiceProvider"
            ],
            "aliases": {
                "Widget": "Laracasts\\Widget\\Facades\\Widget"
            }
        }
    }
}
```
Also, `WidgetServiceProvider` is in the correct namespace:

```
namespace Laracasts\Widget;

use Illuminate\Support\ServiceProvider;

class WidgetServiceProvider extends ServiceProvider
{
...
```

And finally, `php artisan` works fine. I cleared cache, and did `composer dumpautoload`

I just tried to execute the minimum amount of steps (Laravel 8) by creating a new package with the command, and add it via `composer require my/package` and the error is the same. It seems that the symlink isn't working as the file is there (It's created automatically by the command, so not much of an error to make) Just to double clarify, I'm using Windows 10 natively with WSL linux 

Thanks a lot!!

UPDATE:

I published the package anyways in order to see if it could have been something related to symlinks, and the publishing worked like a charm. I'm actually using it by requiring not from the same project but from a github repo, so it makes me think it's an issue with symlinks? As I mentioned, I started using your library after watching a tutorial in Laracasts, where the teacher suggests the library (And with good reason), and it worked just fine for him, using a mac environment (With native Unix like SO)

So, it'll be good to try to investigate why it didn't work at least for other people who uses WSL, but I understand that you can't just try every single environment on earth. Great work!!!!

**Reply by Jeroen-G** [reply]
Your composer.json is missing stuff:
```
{
    "name": "laracasts/widget",
    "description": ":package_description",
    "license": "MIT",
    "homepage": "https://github.com/laracasts/widget",
    "keywords": ["Laravel", "Widget"],
    "require": {
        "illuminate/support": "~8"
    },
    "require-dev": {
        "phpunit/phpunit": "~9.0",
        "orchestra/testbench": "~9"
    },
    "autoload": {
        "psr-4": {
            "Laracasts\\Widget\\": "src/"
        }
    },
    "autoload-dev": {
        "psr-4": {
            "Laracasts\\Widget\\Tests\\": "tests"
        }
    },
    "extra": {
        "laravel": {
            "providers": [
                "Laracasts\\Widget\\WidgetServiceProvider"
            ],
            "aliases": {
                "Widget": "Laracasts\\Widget\\Facades\\Widget"
            }
        }
    },
    "repositories": {
        "laracasts/widget": {
            "type": "path",
            "url": "???",
            "options": {
                "symlink": true
            }
        },
    }
}
```
The URL is something like `/Users/jeroen/projects/packagertest/packages/myvendor/mypackage` (I am on MacOS)

It might be because of Windows. Not sure.
